stan/math/prim/prob/wiener4_lccdf.hpp

```diff
@@ -0,0 +1,454 @@
+#ifndef STAN_MATH_PRIM_PROB_WIENER4_LCCDF_HPP
+#define STAN_MATH_PRIM_PROB_WIENER4_LCCDF_HPP
+
+#include <stan/math/prim/prob/wiener4_lcdf.hpp>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+/**
+ * Log of probability of reaching the upper bound in diffusion process
+ *
+ * @tparam T_a type of boundary
+ * @tparam T_w type of relative starting point
+ * @tparam T_v type of drift rate
+ *
+ * @param a The boundary separation
+ * @param w_value The relative starting point
+ * @param v_value The drift rate
+ * @return log probability to reach the upper bound
+ */
+template <typename T_a, typename T_w, typename T_v>
+inline auto wiener_prob(const T_a& a, const T_v& v_value,
+                        const T_w& w_value) noexcept {
+  using ret_t = return_type_t<T_a, T_w, T_v>;
+  const auto v = -v_value;
+  const auto w = 1 - w_value;
+  if (fabs(v) == 0.0) {
+    return ret_t(log1p(-w));
+  }
+
+  const auto exponent = -2.0 * v * a * (1.0 - w);
+  if (exponent < 0) {
+    return ret_t(log1m_exp(exponent) - log_diff_exp(2 * v * a * w, exponent));
+  } else {
+    return ret_t(log1m_exp(-exponent) - log1m_exp(2 * v * a));
+  }
```

Comment:
I'm kind of confused by these cutpoints. Is this because the derivative is ill defined at certain areas or is this a math optimization

Reply:
Yes, this is a math optimization and should stay.

Reply:
The if statement here could be more more expensive than the ops that are saved. I'd remove all of these and just keep things simple. It also just becomes really hard to read and maintain with all of these if statements in the code

Reply:
Results are more robust when we have this case distinction. They both shall compute the same result, but when `exponent < 0` then the upper case is more robust and when `exponent >=0` the lower case is more robust. We could insert a comment on this to make the case distinction clear.

Reply:
Numerically robust.